File: modules/swagger-core/src/test/scala/converter/HiddenFieldTest.scala
package converter

import converter.models._

import com.wordnik.swagger.converter._
import com.wordnik.swagger.core.util._
import com.wordnik.swagger.model._
import com.wordnik.swagger.annotations._

import java.util.Date

import scala.reflect.BeanProperty

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.scalatest.matchers.ShouldMatchers

@RunWith(classOf[JUnitRunner])
class HiddenFieldTest extends FlatSpec with ShouldMatchers {
  it should "ignore a hidden field" in {
    val model = ModelConverters.read(classOf[ModelWithHiddenFields]).getOrElse(fail("no model found"))
    model.id should be ("ModelWithHiddenFields")
    model.properties.size should be (2)

    val idValue = model.properties("id")
    idValue.`type` should be ("long")
    idValue.required should be (true)

    val nameValue = model.properties("name")
    nameValue.`type` should be ("string")
    nameValue.required should be (true)
  }
}

class ModelWithHiddenFields {
  @BeanProperty
  @ApiModelProperty(required=true)
  var id: Long = _

  @BeanProperty
  @ApiModelProperty(required=true, hidden=false)
  var name: String = _

  @BeanProperty
  @ApiModelProperty(hidden=true)
  var password: String = _
}

File: modules/swagger-core/src/test/java/io/swagger/HiddenFieldTest.java
package io.swagger;

import io.swagger.converter.ModelConverters;
import io.swagger.oas.models.media.IntegerSchema;
import io.swagger.oas.models.media.Schema;
import io.swagger.oas.models.media.StringSchema;
import org.testng.annotations.Test;

import java.util.Map;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertNotNull;
import static org.testng.Assert.assertTrue;

public class HiddenFieldTest {

    @Test(enabled = false, description = "it should ignore a hidden field")
    public void testHiddenField() {
        final Map<String, Schema> models = ModelConverters.getInstance().read(ModelWithHiddenFields.class);

        final Schema model = models.get("ModelWithHiddenFields");
        assertNotNull(model);
        assertEquals(model.getProperties().size(), 2);

        final Schema idValue = (Schema)model.getProperties().get("id");
        assertTrue(idValue instanceof IntegerSchema);
        // TODO
//        assertTrue(idValue.getRequired());

        final Schema nameValue = (Schema)model.getProperties().get("name");
        assertTrue(nameValue instanceof StringSchema);
    }

    class ModelWithHiddenFields {
        @io.swagger.oas.annotations.media.Schema(required = true)
        public Long id = null;

        // TODO
        @io.swagger.oas.annotations.media.Schema(required = true/*, hidden = false*/)
        public String name = null;

        // TODO
        @io.swagger.oas.annotations.media.Schema(/*hidden = true*/)
        public String password = null;
    }
}
